spark-datasource/src/main/scala/com/salesforce/datacloud/spark/HyperResultTable.scala

```diff
@@ -0,0 +1,100 @@
+package com.salesforce.datacloud.spark
+
+import org.apache.spark.sql.connector.catalog.SupportsRead
+import org.apache.spark.sql.types.StructType
+import java.{util => ju}
+import java.time.Duration
+import org.apache.spark.sql.connector.catalog.TableCapability
+import org.apache.spark.sql.connector.read.ScanBuilder
+import org.apache.spark.sql.util.CaseInsensitiveStringMap
+import org.apache.spark.sql.connector.read.Scan
+import org.apache.spark.sql.connector.read.SupportsReportStatistics
+import org.apache.spark.sql.connector.read.Batch
+import org.apache.spark.sql.connector.read.Statistics
+import scala.util.Using
+import org.apache.spark.sql.connector.metric.CustomMetric
+import org.apache.spark.sql.connector.metric.CustomTaskMetric
+
+private case class HyperResultTable(
+    connectionOptions: HyperConnectionOptions,
+    resultSetId: String,
+    schema: StructType
+) extends SupportsRead {
+  override def name(): String = s"hyper_result_set_${resultSetId}"
+
+  override def capabilities(): ju.Set[TableCapability] = {
+    val capabilities = new ju.HashSet[TableCapability]()
+    capabilities.add(TableCapability.BATCH_READ)
+    capabilities
+  }
+
+  override def newScanBuilder(
+      options: CaseInsensitiveStringMap
+  ): ScanBuilder = {
+    val (chunkCount, rowCount) =
+      Using(connectionOptions.createConnection()) { conn =>
+        // TODO XXX: set infinite timeout
+        val queryStatus =
+          conn.waitForResultsProduced(resultSetId, Duration.ofDays(2))
```

Comment:
Should this timeout be configurable? I can't imagine why infinite/2days would be a good thing for a spark query